pika/connection.py

```diff
@@ -963,27 +963,64 @@ def _set_url_ssl_options(self, value):
         """Deserialize and apply the corresponding query string arg
 
         """
-        options = ast.literal_eval(value)
-        if options is None:
+        opts = ast.literal_eval(value)
+        if opts is None:
             if self.ssl_options is not None:
                 raise ValueError(
                     'Specified ssl_options=None URL arg is inconsistent with '
                     'the specified https URL scheme.')
         else:
-            # Convert options to pika.SSLOptions via ssl.SSLSocket()
-            sock = socket.socket()
-            try:
-                ssl_sock = ssl.SSLSocket(sock=sock, **options)
-                try:
-                    self.ssl_options = pika.SSLOptions(
-                        context=ssl_sock.context,
-                        server_hostname=ssl_sock.server_hostname)
-                finally:
-                    ssl_sock.close()
-            finally:
-                sock.close()
-
-
+            # Note: this is the deprecated wrap_socket signature and info:
```

Comment:
@lukebakken - this comment is confusing. We were using `ssl.SSLSocket` previously, not `wrap_socket`. Did you mean that `ssl.SSLSocket` was deprecated?

Reply:
The `ssl.SSLSocket` ctor we were using is deprecated and the old `wrap_socket`, too.